libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/index.tsx

```diff
@@ -1,17 +1,60 @@
 import React from 'react'
 import { useIntl } from 'react-intl'
-import { Box, Text, ModalBase } from '@island.is/island-ui/core'
+import cn from 'classnames'
+import {
+  Box,
+  Text,
+  ModalBase,
+  Logo,
+  SkeletonLoader,
+  Icon,
+  IconMapIcon,
+  Button,
+} from '@island.is/island-ui/core'
 import { signatureModal } from '../../../lib/messages'
-import { FileSignatureStatus, ReducerState } from '../fileSignatureReducer'
-import ModalConditionalContent from './ModalConditionalContent'
-import * as style from './Modal.treat'
+import {
+  FileSignatureStatus,
+  ReducerState,
+  ContentTypes,
+} from '../fileSignatureReducer'
+import * as styles from './Modal.treat'
+import { Colors } from '@island.is/island-ui/theme'
 
 interface SignatureModalProps {
   fileSignatureState: ReducerState
   onClose: () => void
   controlCode: string
 }
 
+const statusVariantStyles: {
+  [key in ContentTypes]: {
+    background: Colors
+    iconColor: Colors
+    icon: IconMapIcon
+  }
+} = {
+  [ContentTypes.ERROR]: {
+    background: 'red100',
+    iconColor: 'red400',
+    icon: 'warning',
+  },
+  [ContentTypes.INFO]: {
+    background: 'blue100',
+    iconColor: 'blue400',
+    icon: 'informationCircle',
+  },
+  [ContentTypes.SUCCESS]: {
+    background: 'blue100',
+    iconColor: 'blue400',
+    icon: 'checkmarkCircle',
+  },
+  [ContentTypes.WARNING]: {
+    background: 'yellow300',
+    iconColor: 'yellow600',
+    icon: 'warning',
+  },
```

Comment:
Can we clean up these `statusVariantStyles` . The way I am reading this is we only use the icon when the type is `Success` and we don't use the warning nor the error style. If that is correct, can we then extend the stylemap for the controlCodeContainer to receive the needed styles for the that  status?  And then hardcode the values for the icon?

Reply:
Good catch! We should definitely change this 💯